dashboard/app/models/user.rb

```diff
@@ -731,7 +731,15 @@ def advertised_scripts
   def in_progress_and_completed_scripts
     backfill_user_scripts if needs_to_backfill_user_scripts?
 
-    user_scripts.compact
+    user_scripts.compact.reject do |user_script|
+      begin
+        user_script.script.nil?
```

Comment:
`UserScript.script` does exactly that under the hood ([link](https://github.com/code-dot-org/code-dot-org/blob/571b0750e50d14148af5beda90e6d621946870b5/dashboard/app/models/user_script.rb#L26-L26)); is it necessary to do so here as well?

```ruby
# user_script.rb:25
  def script
    Script.get_from_cache(script_id)
  end
```